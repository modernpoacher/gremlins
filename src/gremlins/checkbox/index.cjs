require('@babel/register')({
  ignore: [
    /node_modules\/(?!@modernpoacher\/cogs|@modernpoacher\/sprockets|@modernpoacher\/gremlins)/
  ]
})

const debug = require('debug')

const log = debug('@modernpoacher/gremlins/gremlins/textbox')

log('`gremlins` is awake')

const {
  default: component
} = require('./index.jsx')

module.exports = component
