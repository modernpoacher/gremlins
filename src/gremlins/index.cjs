require('@babel/register')({
  ignore: [
    /node_modules\/(?!@modernpoacher\/cogs|@modernpoacher\/gremlins|@modernpoacher\/sprockets)/
  ]
})

const debug = require('debug')

const log = debug('@modernpoacher/gremlins/gremlins')

log('`gremlins` is awake')

/**
 *  Has default export, etc
 */
module.exports = require('./index.jsx')
