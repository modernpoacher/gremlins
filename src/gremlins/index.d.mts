declare module '#gremlins/gremlins' {
  import React from 'react'

  export interface FieldGremlinProps {
    id: string
    name: string
    required: boolean
    disabled: boolean
    readOnly: boolean
    placeholder: string
    onChange: GremlinsTypes.OnChangeType
    fieldRef: object
  }

  export interface ValueGremlinProps extends FieldGremlinProps {
    value: string
    defaultValue: string
    onChange: GremlinsTypes.OnChangeType
  }

  export interface CheckGremlinProps extends FieldGremlinProps {
    checked: boolean
    defaultChecked: boolean
    onClick: GremlinsTypes.OnClickType
  }

  export interface GroupGremlinProps {
    title: string
    onChange: GremlinsTypes.OnChangeType
    groupRef: object
    children: React.JSX.Element | React.JSX.Element[]
  }

  export class FieldGremlin extends React.Component<FieldGremlinProps> {}

  export class ValueGremlin extends React.Component<ValueGremlinProps> {}

  export class CheckGremlin extends React.Component<CheckGremlinProps> {}

  export class GroupGremlin extends React.Component<GroupGremlinProps> {}
}

declare module '@modernpoacher/gremlins/gremlins' {
  export * from '#gremlins/gremlins'
}
