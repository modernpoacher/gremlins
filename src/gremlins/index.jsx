/**
 * Gremlin component
 */
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import Field from '@modernpoacher/gremlins/components/field'

function onChange () {
  /* */
}

function onClick () {
  /* */
}

export default class Gremlin extends Component {
  getClassName () {
    const {
      required,
      disabled,
      readOnly
    } = this.props

    return classnames('gremlin', { required, disabled, readOnly })
  }

  getId () {
    const {
      id,
      name
    } = this.props

    return id || name
  }

  shouldComponentUpdate (props) {
    return (
      (props.name !== this.props.name) ||
      (props.id !== this.props.id) ||
      (props.required !== this.props.required) ||
      (props.disabled !== this.props.disabled) ||
      (props.readOnly !== this.props.readOnly) ||
      (props.tabIndex !== this.props.tabIndex) ||
      (props.accessKey !== this.props.accessKey) ||
      (props.placeholder !== this.props.placeholder) ||
      (props.onChange !== this.props.onChange)
    )
  }

  renderField () {
    const id = this.getId()

    const {
      name,
      required,
      disabled,
      readOnly,
      tabIndex,
      accessKey,
      placeholder,
      onChange,
      fieldRef
    } = this.props

    return (
      <Field
        id={id}
        name={name}
        required={required}
        disabled={disabled}
        readOnly={readOnly}
        tabIndex={tabIndex}
        accessKey={accessKey}
        placeholder={placeholder}
        onChange={onChange}
        fieldRef={fieldRef}
      />
    )
  }

  render () {
    const className = this.getClassName()

    return (
      <div className={className}>
        {this.renderField()}
      </div>
    )
  }
}

Gremlin.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
  required: PropTypes.bool,
  disabled: PropTypes.bool,
  readOnly: PropTypes.bool,
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
  fieldRef: PropTypes.shape({
    current: PropTypes.shape().isRequired
  })
}

Gremlin.defaultProps = {
  required: false,
  disabled: false,
  readOnly: false,
  onChange
}

export class ValueGremlin extends Gremlin {
  shouldComponentUpdate (props) {
    return (
      super.shouldComponentUpdate(props) ||
      (props.value !== this.props.value)
    )
  }
}

ValueGremlin.propTypes = {
  ...Gremlin.propTypes,
  value: PropTypes.string,
  defaultValue: PropTypes.string
}

ValueGremlin.defaultProps = {
  ...Gremlin.defaultProps
}

export class CheckGremlin extends Gremlin {
  shouldComponentUpdate (props) {
    return (
      super.shouldComponentUpdate(props) ||
      (props.checked !== this.props.checked) ||
      (props.onClick !== this.props.onClick)
    )
  }
}

CheckGremlin.propTypes = {
  ...Gremlin.propTypes,
  checked: PropTypes.bool,
  defaultChecked: PropTypes.bool,
  onClick: PropTypes.func
}

CheckGremlin.defaultProps = {
  ...Gremlin.defaultProps,
  onClick
}
