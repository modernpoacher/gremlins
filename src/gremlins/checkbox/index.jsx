/**
 * CheckboxGremlin component
 */
import React from 'react'
import classnames from 'classnames'

import { CheckGremlin } from '@modernpoacher/gremlins/gremlins'

import Field from './field/index.jsx'

export default class CheckboxGremlin extends CheckGremlin {
  getClassName () {
    return classnames(super.getClassName(), 'checkbox')
  }

  handleClick = (value) => {
    const {
      onClick
    } = this.props

    onClick(value)
  }

  handleChange = (value) => {
    const {
      onChange
    } = this.props

    onChange(value)
  }

  renderField () {
    const id = this.getId()

    const {
      name,
      checked,
      defaultChecked,
      required,
      disabled,
      readOnly,
      tabIndex,
      accessKey,
      placeholder,
      fieldRef
    } = this.props

    return (
      <Field
        id={id}
        name={name}
        checked={checked}
        defaultChecked={defaultChecked}
        required={required}
        disabled={disabled}
        readOnly={readOnly}
        tabIndex={tabIndex}
        accessKey={accessKey}
        placeholder={placeholder}
        onClick={this.handleClick}
        onChange={this.handleChange}
        fieldRef={fieldRef}
      />
    )
  }

  render () {
    const className = this.getClassName()

    return (
      <div className={className}>
        {this.renderField()}
      </div>
    )
  }
}

CheckboxGremlin.propTypes = {
  ...CheckGremlin.propTypes
}

CheckboxGremlin.defaultProps = {
  ...CheckGremlin.defaultProps
}
