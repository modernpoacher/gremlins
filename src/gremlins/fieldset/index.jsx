/**
 * FieldsetGremlin component
 *
 * @typedef {import('@modernpoacher/gremlins/gremlins').GroupGremlinProps} GroupGremlinProps
 */

import React from 'react'

import classnames from 'classnames'

import { GroupGremlin as Gremlin } from '@modernpoacher/gremlins/gremlins'
import Group from './group/index.jsx'

export default class FieldsetGremlin extends Gremlin {
  getClassName () {
    return classnames(super.getClassName(), 'fieldset')
  }

  renderGroup () {
    const {
      onChange,
      children
    } = this.props

    return (
      <Group
        onChange={onChange}>
        {children}
      </Group>
    )
  }
}

FieldsetGremlin.propTypes = {
  ...Gremlin.propTypes
}

FieldsetGremlin.defaultProps = {
  ...Gremlin.defaultProps
}
