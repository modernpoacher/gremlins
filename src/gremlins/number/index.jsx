/**
 * NumberGremlin component
 */
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import { ValueGremlin } from '@modernpoacher/gremlins/gremlins'

import Field from './field/index.jsx'

export default class NumberGremlin extends ValueGremlin {
  getClassName () {
    return classnames(super.getClassName(), 'number')
  }

  handleChange = (value) => {
    const {
      onChange
    } = this.props

    onChange(value)
  }

  renderField () {
    const id = this.getId()

    const {
      name,
      value,
      defaultValue,
      required,
      disabled,
      readOnly,
      tabIndex,
      accessKey,
      placeholder,
      fieldRef
    } = this.props

    return (
      <Field
        id={id}
        name={name}
        value={value}
        defaultValue={defaultValue}
        required={required}
        disabled={disabled}
        readOnly={readOnly}
        tabIndex={tabIndex}
        accessKey={accessKey}
        placeholder={placeholder}
        onChange={this.handleChange}
        fieldRef={fieldRef}
      />
    )
  }
}

NumberGremlin.propTypes = {
  ...ValueGremlin.propTypes,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  defaultValue: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ])
}

NumberGremlin.defaultProps = {
  ...ValueGremlin.defaultProps
}
