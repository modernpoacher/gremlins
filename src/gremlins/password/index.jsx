/**
 * PasswordGremlin component
 */
import React from 'react'
import classnames from 'classnames'

import { ValueGremlin } from '@modernpoacher/gremlins/gremlins'

import Field from './field/index.jsx'

export default class PasswordGremlin extends ValueGremlin {
  getClassName () {
    return classnames(super.getClassName(), 'password')
  }

  handleChange = (value) => {
    const {
      onChange
    } = this.props

    onChange(value)
  }

  renderField () {
    const id = this.getId()

    const {
      name,
      value,
      defaultValue,
      required,
      disabled,
      readOnly,
      tabIndex,
      accessKey,
      placeholder,
      fieldRef
    } = this.props

    return (
      <Field
        id={id}
        name={name}
        value={value}
        defaultValue={defaultValue}
        required={required}
        disabled={disabled}
        readOnly={readOnly}
        tabIndex={tabIndex}
        accessKey={accessKey}
        placeholder={placeholder}
        onChange={this.handleChange}
        fieldRef={fieldRef}
      />
    )
  }
}

PasswordGremlin.propTypes = {
  ...ValueGremlin.propTypes
}

PasswordGremlin.defaultProps = {
  ...ValueGremlin.defaultProps
}
