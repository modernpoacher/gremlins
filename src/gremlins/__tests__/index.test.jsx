import React from 'react'
import renderer from 'react-test-renderer'
import classnames from 'classnames'

import Field from '#gremlins/components/field'

import { FieldGremlin as Gremlin, ValueGremlin, CheckGremlin } from '#gremlins/gremlins'

jest.mock('classnames', () => jest.fn(() => 'MOCK CLASSNAME'))

jest.mock('#gremlins/components/field')

describe('#gremlins/gremlins', () => {
  describe('<Gremlin />', () => {
    describe('With required props', () => {
      const component = (
        <Gremlin name='MOCK NAME' />
      )

      it('renders', () => {
        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })

      describe('Prototype', () => {
        describe('`getClassName`', () => {
          it('is defined', () => {
            return expect(Gremlin.prototype.getClassName)
              .toBeDefined()
          })
        })

        describe('`getId`', () => {
          it('is defined', () => {
            return expect(Gremlin.prototype.getId)
              .toBeDefined()
          })
        })

        describe('`shouldComponentUpdate`', () => {
          it('is defined', () => {
            return expect(Gremlin.prototype.shouldComponentUpdate)
              .toBeDefined()
          })
        })

        describe('`renderField`', () => {
          it('is defined', () => {
            return expect(Gremlin.prototype.renderField)
              .toBeDefined()
          })
        })
      })
    })

    describe('With additional props', () => {
      it('renders', () => {
        const component = (
          <Gremlin
            name='MOCK NAME'
            id='MOCK ID'
            tabIndex={1}
            accessKey='MOCK ACCESS KEY'
            required
            disabled
            readOnly
            placeholder='MOCK PLACEHOLDER'
            onChange={jest.fn()}
          />
        )

        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })
    })

    describe('`getClassName()`', () => {
      beforeEach(() => {
        classnames.mockReset()
      })

      describe('With required props', () => {
        let returnValue

        beforeEach(() => {
          classnames.mockReturnValue('MOCK CLASSNAME')

          const component = (
            <Gremlin name='MOCK NAME' />
          )

          const instance = (
            renderer.create(component)
              .getInstance()
          )

          returnValue = instance.getClassName()
        })

        it('invokes `classnames`', () => {
          return expect(classnames)
            .toBeCalledWith('gremlin', { required: false, disabled: false, readOnly: false })
        })

        it('returns the classname', () => {
          return expect(returnValue)
            .toBe('MOCK CLASSNAME')
        })
      })

      describe('With additional props', () => {
        describe('`props` has `required`', () => {
          let returnValue

          beforeEach(() => {
            classnames.mockReturnValue('MOCK CLASSNAME')

            const component = (
              <Gremlin name='MOCK NAME' required />
            )

            const instance = (
              renderer.create(component)
                .getInstance()
            )

            returnValue = instance.getClassName()
          })

          it('invokes `classnames`', () => {
            return expect(classnames)
              .toBeCalledWith('gremlin', { required: true, disabled: false, readOnly: false })
          })

          it('returns the classname', () => {
            return expect(returnValue)
              .toBe('MOCK CLASSNAME')
          })
        })

        describe('`props` has `disabled`', () => {
          let returnValue

          beforeEach(() => {
            classnames.mockReturnValue('MOCK CLASSNAME')

            const component = (
              <Gremlin name='MOCK NAME' disabled />
            )

            const instance = (
              renderer.create(component)
                .getInstance()
            )

            returnValue = instance.getClassName()
          })

          it('invokes `classnames`', () => {
            return expect(classnames)
              .toBeCalledWith('gremlin', { required: false, disabled: true, readOnly: false })
          })

          it('returns the classname', () => {
            return expect(returnValue)
              .toBe('MOCK CLASSNAME')
          })
        })

        describe('`props` has `readOnly`', () => {
          let returnValue

          beforeEach(() => {
            classnames.mockReturnValue('MOCK CLASSNAME')

            const component = (
              <Gremlin name='MOCK NAME' readOnly />
            )

            const instance = (
              renderer.create(component)
                .getInstance()
            )

            returnValue = instance.getClassName()
          })

          it('invokes `classnames`', () => {
            return expect(classnames)
              .toBeCalledWith('gremlin', { required: false, disabled: false, readOnly: true })
          })

          it('returns the classname', () => {
            return expect(returnValue)
              .toBe('MOCK CLASSNAME')
          })
        })
      })
    })

    describe('`getId()`', () => {
      describe('With required props', () => {
        describe('`props` has `name`', () => {
          it('returns the id', () => {
            const component = (
              <Gremlin name='MOCK NAME' />
            )

            return expect(renderer.create(component).getInstance().getId())
              .toBe('MOCK NAME')
          })
        })
      })

      describe('With additional props', () => {
        describe('`props` has `id`', () => {
          it('returns the id', () => {
            const component = (
              <Gremlin name='MOCK NAME' id='MOCK ID' />
            )

            return expect(renderer.create(component).getInstance().getId())
              .toBe('MOCK ID')
          })
        })
      })
    })

    describe('`shouldComponentUpdate()`', () => {
      const MOCK_ON_CHANGE = jest.fn()

      const component = (
        <Gremlin
          name='MOCK NAME'
          id='MOCK ID'
          tabIndex={1}
          accessKey='MOCK ACCESS KEY'
          required
          disabled
          readOnly
          placeholder='MOCK PLACEHOLDER'
          onChange={MOCK_ON_CHANGE}
        />
      )

      let instance

      beforeEach(() => {
        instance = renderer.create(component).getInstance()
      })

      describe('`props` have changed', () => {
        it('returns true', () => {
          return expect(instance.shouldComponentUpdate({
            name: 'MOCK CHANGE NAME',
            id: 'MOCK CHANGE ID',
            tabIndex: 0,
            accessKey: 'MOCK CHANGE ACCESS KEY',
            required: false,
            disabled: false,
            readOnly: false,
            placeholder: 'MOCK CHANGE PLACEHOLDER',
            onChange: jest.fn()
          }))
            .toBe(true)
        })
      })

      describe('`props` have not changed', () => {
        it('returns false', () => {
          return expect(instance.shouldComponentUpdate({ // instance.props
            name: 'MOCK NAME',
            id: 'MOCK ID',
            tabIndex: 1,
            accessKey: 'MOCK ACCESS KEY',
            required: true,
            disabled: true,
            readOnly: true,
            placeholder: 'MOCK PLACEHOLDER',
            onChange: MOCK_ON_CHANGE
          }))
            .toBe(false)
        })
      })
    })

    describe('`renderField()`', () => {
      const component = (
        <Gremlin
          name='MOCK NAME'
          id='MOCK ID'
          tabIndex={1}
          accessKey='MOCK ACCESS KEY'
          required
          disabled
          readOnly
          placeholder='MOCK PLACEHOLDER'
        />
      )

      let instance

      let getIdSpy

      beforeEach(() => {
        jest.clearAllMocks()

        instance = (
          renderer.create(component)
            .getInstance()
        )

        getIdSpy = jest.spyOn(Gremlin.prototype, 'getId').mockReturnValue('MOCK ID')

        instance.renderField()
      })

      it('invokes `getId`', () => {
        return expect(getIdSpy)
          .toBeCalled()
      })

      it('renders `<Field />`', () => {
        return expect(Field)
          .toBeCalledWith({
            name: 'MOCK NAME',
            id: 'MOCK ID',
            tabIndex: 1,
            accessKey: 'MOCK ACCESS KEY',
            required: true,
            disabled: true,
            readOnly: true,
            placeholder: 'MOCK PLACEHOLDER',
            onChange: expect.any(Function)
          }, {})
      })
    })
  })

  describe('<ValueGremlin />', () => {
    describe('With required props', () => {
      const component = (
        <ValueGremlin name='MOCK NAME' />
      )

      it('renders', () => {
        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })

      describe('Prototype', () => {
        describe('`shouldComponentUpdate`', () => {
          it('is defined', () => {
            return expect(ValueGremlin.prototype.shouldComponentUpdate)
              .toBeDefined()
          })
        })
      })

      describe('Instance', () => {
        describe('Extends `<Gremlin />`', () => {
          it('is defined', () => {
            return expect(renderer.create(component).getInstance())
              .toBeInstanceOf(Gremlin)
          })
        })
      })
    })

    describe('With additional props', () => {
      it('renders', () => {
        const component = (
          <ValueGremlin
            name='MOCK NAME'
            id='MOCK ID'
            tabIndex={1}
            accessKey='MOCK ACCESS KEY'
            required
            disabled
            readOnly
            value='MOCK VALUE'
            placeholder='MOCK PLACEHOLDER'
            onChange={jest.fn()}
          />
        )

        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })
    })

    describe('`shouldComponentUpdate()`', () => {
      const MOCK_ON_CHANGE = jest.fn()

      const component = (
        <ValueGremlin
          name='MOCK NAME'
          id='MOCK ID'
          tabIndex={1}
          accessKey='MOCK ACCESS KEY'
          required
          disabled
          readOnly
          value='MOCK VALUE'
          placeholder='MOCK PLACEHOLDER'
          onChange={MOCK_ON_CHANGE}
        />
      )

      let instance

      beforeEach(() => {
        instance = renderer.create(component).getInstance()
      })

      describe('`props` have changed', () => {
        it('returns true', () => {
          return expect(instance.shouldComponentUpdate({
            ...instance.props,
            value: 'MOCK CHANGE VALUE',
            onChange: jest.fn()
          }))
            .toBe(true)
        })
      })

      describe('`props` have not changed', () => {
        it('returns false', () => {
          return expect(instance.shouldComponentUpdate({
            ...instance.props,
            value: 'MOCK VALUE',
            onChange: MOCK_ON_CHANGE
          }))
            .toBe(false)
        })
      })
    })
  })

  describe('<CheckGremlin />', () => {
    describe('With required props', () => {
      const component = (
        <CheckGremlin name='MOCK NAME' />
      )

      it('renders', () => {
        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })

      describe('Prototype', () => {
        describe('`shouldComponentUpdate`', () => {
          it('is defined', () => {
            return expect(CheckGremlin.prototype.shouldComponentUpdate)
              .toBeDefined()
          })
        })
      })

      describe('Instance', () => {
        describe('Extends `<Gremlin />`', () => {
          it('is defined', () => {
            return expect(renderer.create(component).getInstance())
              .toBeInstanceOf(Gremlin)
          })
        })
      })
    })

    describe('With additional props', () => {
      it('renders', () => {
        const component = (
          <CheckGremlin
            name='MOCK NAME'
            id='MOCK ID'
            tabIndex={1}
            accessKey='MOCK ACCESS KEY'
            required
            disabled
            readOnly
            checked
            placeholder='MOCK PLACEHOLDER'
            onChange={jest.fn()}
            onClick={jest.fn()}
          />
        )

        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })
    })

    describe('`shouldComponentUpdate()`', () => {
      const MOCK_ON_CHANGE = jest.fn()

      const component = (
        <CheckGremlin
          name='MOCK NAME'
          id='MOCK ID'
          tabIndex={1}
          accessKey='MOCK ACCESS KEY'
          required
          disabled
          readOnly
          checked
          placeholder='MOCK PLACEHOLDER'
          onChange={MOCK_ON_CHANGE}
        />
      )

      let instance

      beforeEach(() => {
        instance = renderer.create(component).getInstance()
      })

      describe('`props` have changed', () => {
        it('returns true', () => {
          return expect(instance.shouldComponentUpdate({
            ...instance.props,
            checked: false
          }))
            .toBe(true)
        })
      })

      describe('`props` have not changed', () => {
        it('returns false', () => {
          return expect(instance.shouldComponentUpdate({ // instance.props
            ...instance.props,
            checked: true
          }))
            .toBe(false)
        })
      })
    })
  })
})
