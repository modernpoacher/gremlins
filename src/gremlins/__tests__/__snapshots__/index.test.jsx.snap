// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@modernpoacher/gremlins/gremlins <CheckGremlin /> With additional props renders 1`] = `
<div
  className="MOCK CLASSNAME"
/>
`;

exports[`@modernpoacher/gremlins/gremlins <CheckGremlin /> With required props renders 1`] = `
<div
  className="MOCK CLASSNAME"
/>
`;

exports[`@modernpoacher/gremlins/gremlins <Gremlin /> With additional props renders 1`] = `
<div
  className="MOCK CLASSNAME"
/>
`;

exports[`@modernpoacher/gremlins/gremlins <Gremlin /> With required props renders 1`] = `
<div
  className="MOCK CLASSNAME"
/>
`;

exports[`@modernpoacher/gremlins/gremlins <ValueGremlin /> With additional props renders 1`] = `
<div
  className="MOCK CLASSNAME"
/>
`;

exports[`@modernpoacher/gremlins/gremlins <ValueGremlin /> With required props renders 1`] = `
<div
  className="MOCK CLASSNAME"
/>
`;
